packages/html-reporter/src/testResultView.tsx

```diff
@@ -26,57 +26,45 @@ import { AttachmentLink } from './links';
 import { statusIcon } from './statusIcon';
 import './testResultView.css';
 
-type DiffTab = {
-  id: string,
-  title: string,
-  attachment: TestAttachment,
+type ImageDiff = {
+  name: string,
+  left?: { attachment: TestAttachment, title: string },
+  right?: { attachment: TestAttachment, title: string },
+  diff?: { attachment: TestAttachment, title: string },
 };
 
-function classifyAttachments(attachments: TestAttachment[]) {
-  const screenshots = new Set(attachments.filter(a => a.contentType.startsWith('image/')));
-  const videos = attachments.filter(a => a.name === 'video');
-  const traces = attachments.filter(a => a.name === 'trace');
-
-  const otherAttachments = new Set<TestAttachment>(attachments);
-  [...screenshots, ...videos, ...traces].forEach(a => otherAttachments.delete(a));
-
-  const snapshotNameToDiffTabs = new Map<string, DiffTab[]>();
-  let tabId = 0;
-  for (const attachment of attachments) {
-    const match = attachment.name.match(/^(.*)-(\w+)(\.[^.]+)?$/);
+function groupImageDiffs(screenshots: Set<TestAttachment>): ImageDiff[] {
```

Comment:
Maybe rename since `screenshots` is actually an out parameter?

```suggestion
function extractImageDiffs(screenshots: Set<TestAttachment>): ImageDiff[] {
```

Reply:
will follow up!